quickscreen/analysis/datafill.py

```diff
@@ -98,12 +109,17 @@ def __sub__(self, other):
         --------
         >>> difference = DataEdit(df1) - df2
         """
-        assert isinstance(other, pd.core.frame.DataFrame), "Not a Pandas data frame."
         try:
-            common = self.data.merge(other, indicator='i', how='outer').query('i=="left_only"').drop('i',1)
+            assert isinstance(other, pd.DataFrame)
+            try:
+                common = self.data.merge(other, indicator='i', how='outer').query('i=="left_only"').drop('i',1)
+            except:
+                common = self.data
+            return DataEdit(common)
         except:
-            common = self.data
-        return DataEdit(common)
+            print("Not a Pandas data frame.")
+            return
+
 
```

Comment:
same comment about return and nested try statements are maybe not ideal - is the inner one needed?

Reply:
yeah, this seems a little funky

I think it could be replaced with

try:
    statement0
except ExceptionTypeOne:
    statement1
except ExceptionTypeTwo:
    statement2